src/frontend/apps/desk/src/features/mail-domains/components/__tests__/ModalCreateMailbox.test.tsx

```diff
@@ -0,0 +1,327 @@
+import { ModalProvider } from '@openfun/cunningham-react';
+import {
+  QueryClient,
+  QueryClientProvider,
+  useMutation,
+} from '@tanstack/react-query';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { APIError } from '@/api';
+
+import { CreateMailboxParams } from '../../api';
+import { MailDomain } from '../../types';
+import { ModalCreateMailbox } from '../ModalCreateMailbox';
+
+const mockMailDomain: MailDomain = {
+  name: 'domain.fr',
+  id: '456ac6ca-0402-4615-8005-69bc1efde43f',
+  created_at: new Date().toISOString(),
+  updated_at: new Date().toISOString(),
+  slug: 'domainfr',
+  status: 'enabled',
+  abilities: {
+    get: true,
+    patch: true,
+    put: true,
+    post: true,
+    delete: true,
+    manage_accesses: true,
+  },
+};
+
+jest.mock('react-i18next', () => ({
+  useTranslation: () => ({
+    t: (key: string) => key,
+  }),
+}));
+
+jest.mock('@openfun/cunningham-react', () => ({
+  ...jest.requireActual('@openfun/cunningham-react'),
+  useToastProvider: () => ({
+    toast: jest.fn(),
+  }),
+}));
+
+const mockCreateMailbox = jest.fn();
+const mockOnSuccess = jest.fn();
+jest.mock('../../api/useCreateMailbox', () => ({
+  useCreateMailbox: jest.fn().mockImplementation(() => {
+    return useMutation<void, APIError, CreateMailboxParams>({
+      mutationFn: mockCreateMailbox,
+      onSuccess: mockOnSuccess,
+    });
+  }),
+}));
```

Comment:
To test api with jest we use `fetchMock`.

You have some examples in this test
https://github.com/numerique-gouv/people/blob/main/src/frontend/apps/desk/src/features/teams/member-management/__tests__/MemberGrid.test.tsx#L79-L82